packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -304,7 +304,9 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
                 }
                 exist = await this.fileSystem.exists(selected.toString());
                 if (exist) {
-                    overwrite = await this.confirmOverwrite(selected); // TODO this should be handled differently in Electron.
+                    // Prompt users for confirmation before overwriting.
+                    // Electron handles this case itself, so do not prompt again.
+                    overwrite = (environment.electron.is()) ? true : await this.confirmOverwrite(selected);
```

Comment:
`electron.is` will cause problems, unfortunately, when you start a browser Theia from the electron Theia app. See here: https://github.com/theia-ide/theia/issues/4288
Can we make this logic somehow part of the file dialog service? I do not know if it is possible though :/

Reply:
Please note, getting rid of the extension manager might happen soon, so perhaps we just have to wait a bit before you start investigating the file dialog service.

Reply:
Okay sounds good! I wanted to try and fix this as not to block #4771

Reply:
> Okay sounds good!

I am a bit puzzled; so what should I do? Review this, wait, or nothing?

Reply:
Sorry for the confusion. I assume if you believe it will cause errors then we can leave it?
However, there are many instances in the same file where we check for `isElectron()`, and I'm not sure that starting a browser instance from Electron is a common use case outside of Theia developers.

Reply:
> and I'm not sure that starting a browser instance from Electron is a common use case outside of Theia developers.

Understood, I won't hold this PR back but then it would be great if you could close this PR and integrate your changes into #4771. In the end, you're fixing this for the new `Save As...` feature. Thoughts?

Reply:
Sure I can do that :) I'll be fixing both the `Save As...` and `Save Workspace As...`